ResearchKit/Common/ORKOrderedTask.m

```diff
@@ -1090,24 +1091,22 @@ + (ORKOrderedTask *)timedWalkTaskWithIdentifier:(NSString *)identifier
             
             ORKStepArrayAddStep(steps, step);
         }
-        
+
+        {
+            ORKTimedWalkStep *step = [[ORKTimedWalkStep alloc] initWithIdentifier:ORKTimedWalkTurnAroundStepIdentifier];
+            step.title = ORKLocalizedString(@"TIMED_WALK_INSTRUCTION_TURN", nil);
+            step.text = ORKLocalizedString(@"TIMED_WALK_INSTRUCTION_TEXT", nil);
+            step.spokenInstruction = step.title;
+            step.recorderConfigurations = recorderConfigurations;
+            step.distanceInMeters = 1;
+            step.shouldTintImages = YES;
+            step.image = [UIImage imageNamed:@"turnaround" inBundle:[NSBundle bundleForClass:[self class]] compatibleWithTraitCollection:nil];
+            step.stepDuration = timeLimit == 0 ? CGFLOAT_MAX : timeLimit;
```

Comment:
@YuanZhu-apple That's a very good question; I'd say it's difficult to give a general answer. From our experience, yes, it's usually shorter to turn around; but for some people it's actually harder and takes more time, compared to just walking in a straight line.

So it's difficult to define a good constant for that. We could always add it as a parameter, but that would overcomplicate this task, IMO.